### ISSUE TITLE
get a warning when use snapshot in a function component

### ISSUE THREAD

**Issue by sansarun** [pmndrs/eslint-plugin-valtio#22]
This code gives a warning `Better to just use proxy state.eslint(valtio/state-snapshot-rule)`

``` javascript
const MyComponent = () => {
  const snap = useSnapshot(state);
  return <div>{snap.count}</div>;
};
```

It happens only when declare a component using an arrow function.
No warning when declare using `function` keyword.

I guess it's because the function component is detected as a callback.

**Comment by dai-shi** [comment]
Thanks for reporting...
@barelyhuman Seems like we forgot this in tests??

**Comment by barelyhuman** [comment]
I’ll add one, will fix it if it’s being conflicted with one of the previous changes

**Comment by barelyhuman** [comment]
so, it's a conflict between the original implementation and the dependency check we added, will get this fixed ASAP, might take me a day or so, I don't have access to my laptop right now.
@dai-shi

**Comment by dai-shi** [comment]
Published: https://www.npmjs.com/package/eslint-plugin-valtio/v/0.4.4
